src/concurrent_files.rs

```diff
@@ -0,0 +1,281 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::thread;
+
+use crossbeam::channel::{unbounded, Receiver, Sender};
+use globset::GlobSet;
+use walkdir::{DirEntry, WalkDir};
+
+type ProcFiles<Config> = Box<dyn Fn(PathBuf, &Config) -> std::io::Result<()> + Send + Sync>;
```

Comment:
This is only used in `ConcurrentRunner` definition and you use the same name (i.e. `ProcFiles`) here:
https://github.com/mozilla/rust-code-analysis/pull/709/files#diff-8c879faf238b8da3dbf21276b270b4832682947ff73ad1a1607a1371f89f66daR29
which is a bit confusing.
Wouldn't it be simpler to use:
`type ProcFiles = Fn(PathBuf, &Config) -> std::io::Result<()> + Send + Sync;`
with a better name to remember that's a function type ?